core/utils.c

```diff
@@ -2546,7 +2546,7 @@ strcasecmp_with_wildcards(const char *regexp, const char *consider)
             return -1;
         } else if (*consider == '\0')
             return 1;
-        ASSERT(*regexp != EOF && *consider != EOF);
+        ASSERT((sbyte)*regexp != EOF && (sbyte)*consider != EOF);
```

Comment:
Wondering why this change was required.

In any case, I think EOF is an int (https://www.gnu.org/software/libc/manual/html_node/EOF-and-Errors.html), so also wondering why we cast the other operand to an `sbyte`

Reply:
EOF is defined as -1.  *regexp is `char` which is unsigned so if it's `0xff` and we cast to `int` it becomes 255, not -1.  So we have to first cast to a signed type before widening.